spacy_llm/pipeline/llm.py

```diff
@@ -135,16 +146,32 @@ def pipe(self, stream: Iterable[Doc], *, batch_size: int = 128) -> Iterator[Doc]
             try:
                 prompts = self._task.generate_prompts(noncached_doc_batch)
                 responses = self._backend(prompts)
+
+                if self._save_io:
+                    prompts, iter_prompts = tee(prompts)
+                    responses, iter_responses = tee(responses)
+
                 modified_docs = iter(
                     self._task.parse_responses(noncached_doc_batch, responses)
                 )
+
+                if self._save_io:
+                    iter_prompts = iter(prompts)
+                    iter_responses = iter(responses)
+
```

Comment:
And then you don't need this.

Reply:
Right, I saw that... I pushed the necessary commits.